src/components/questions/addAnswer.js

```diff
@@ -0,0 +1,147 @@
+import React from "react";
+import axios from "axios";
+
+class AddAnswer extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      qID : this.props.qID,
+      answer:"",
+      nickname:"",
+      email:"",
+      requiredAnswer: false,
+      requiredNickname: false,
+      requiredEmail: false,
+      validEmail: false
+    }
+    this.onChange = this.onChange.bind(this);
+    this.submitAnswer = this.submitAnswer.bind(this);
+    this.checkEmailValidity = this.checkEmailValidity.bind(this);
+  }
+
+  onChange(event){
+    console.log(event.target.value);
+    let feild = event.target.name;
+    let value = event.target.value;
+    this.setState({[feild] : value})
+  }
+
+  checkEmailValidity(email){
+    let regex = RegExp(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/);
+    return regex.test(email)
+  }
+
+  submitAnswer(event){
+    // event.preventDefault()
+    if (this.state.answer.length === 0) {
+      this.setState({requiredAnswer : true});
+    } else if (this.state.nickname.length === 0) {
+      this.setState({requiredNickname : true});
+    } else if (this.state.email.length === 0) {
+      this.setState({requiredEmail : true});
+    } else if (!this.checkEmailValidity(this.state.email)){
+      this.setState({validEmail : true});
+    } else {
+      axios({
+        method: "POST",
+        url: `http://3.134.102.30/qa/${this.state.qID}/answers`,
+        data:{
+          body:this.state.answer,
+          name:this.state.nickname,
+          email:this.state.email
+        }
+      })
+      .then(data=>{
+        console.log(data);
+        console.log(this.state.qID);
+        console.log("*** Successfully Posted Answer to Question ***")
+        this.props.getProductQuestions(this.props.paramsId);
+        this.props.exitClick()
+      })
+      .catch(err=>{
+        // console.error(err);
+        console.error("*** Error Posting Answer to Question ***")
+      })
+    }
+  }
+
+  render(){
+    return (
+    <React.Fragment>
+
+
+      <label>Your Answer*</label>
+      <br/>
+      <textarea type="text"
+                name="answer"
+                rows="15"
+                cols="70"
+                maxLength="1000"
+                onChange={this.onChange}
+                placeholder="Your Answer"
+      ></textarea>
+      <br/>
+      {/* DISPLAY WARNING IF ANSWER IS GETTING TOO LONG*/}
+      {this.state.answer.length > 998 ?
+      <React.Fragment><div>Your Answer is Too Long! Please Shorten</div><br/></React.Fragment> :
```

Comment:
Is max length 1000? Maybe rephrase this to say it is close to being too large.

Reply:
Max length is 1000 characters but it wont allow input of more than 1000 with the "maxLength" property set so i just set the message to display on 999 as a work around